packages/apollo/src/commands/service/delete.ts

```diff
@@ -34,9 +34,9 @@ export default class ServiceDelete extends ProjectCommand {
             throw new Error("No service found to link to Engine");
           }
 
-          if (!flags.federated) {
+          if (!flags.federated && !flags.serviceName) {
             this.error(
-              "Deleting a service is only supported for federated services. Use the --federated flag if this is a federated service."
+              "Deleting a service is only supported for federated services. Use the --serviceName flag if this is a federated service."
```

Comment:
This entire condition shouldn't be possible given that it's a `required` flag. The CLI framework will throw if the flag isn't provided. If we need it for the sake of making TS happy, we can leave it in - but a TODO comment to remove it altogether would be great (should just need to get the types correct).